### PR TITLE
Add test for value_type alias in vector_api

### PR DESCRIPTION
**Pull request by lbushi25**
This PR adds testing logic to verify that the type described by the vec::value_type alias is the same as the underlying type of the vector.

### CODE REVIEW

File: .github/workflows/cts_ci.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
       matrix:
         include:
           - sycl-impl: dpcpp
-            version: 1dbee22f9c8a3a825deb871bab76937e04fa26fc
+            version: 33e5b101fadc4259135231f99ab62d077b284810
```
Comment (keryell) [inline .github/workflows/cts_ci.yml:59]: Is this related to the issue?

**Review by keryell — APPROVED** [review]
Good catch!

File: .github/workflows/cts_ci.yml (hunk 1 above)
Comment (bader) [reply .github/workflows/cts_ci.yml:59]: @keryell, DPC++ version uplift is required to pass updated test. The version we use today doesn't pass as you can see in pre-commit results.

@lbushi25, DPC++ version uplift can be validated only for the local branches. I'll open a new PR using a local branch to validate the uplift status.

Comment (steffenlarsen) [reply .github/workflows/cts_ci.yml:59]: https://github.com/KhronosGroup/SYCL-CTS/pull/880 should carry the required changes to intel/llvm DPCPP.

**Comment by bader** [comment]
@lbushi25, please, update your local branch to make sure that #880 is there.

**Comment by lbushi25** [comment]
> @lbushi25, please, update your local branch to make sure that #880 is there.

Thanks for the heads up, I went ahead and updated it!

**Comment by lbushi25** [comment]
@bader ping for merge!
